pkg/helm/chartproxy/repos_test.go

```diff
@@ -335,9 +344,19 @@ func TestHelmRepo_IndexFile(t *testing.T) {
 				if err != nil {
 					t.Error(err)
 				}
-				if !reflect.DeepEqual(expectedIndex.Entries, index.Entries) {
+				if !tt.resultDiffers && !reflect.DeepEqual(expectedIndex.Entries, index.Entries) {
```

Comment:
see the comment above: we can revert this change, and create `expectedIndex` as following:

```go
var expectedIndex *repo.IndexFile
if (tt.expectedIndexFile != "") {
  expectedIndex, err = helmrepo.LoadIndexFile(tt.expectedIndexFile)
} else {
  expectedIndex, err = helmrepo.LoadIndexFile(tt.indexFile)
}
```